### PR TITLE
feat: Add delete market to keeper, for future proofing

### PR DESCRIPTION
**Pull request by Eric-Warehime**
Add delete method to x/mm keeper, so that in the future we can use this to remove markets in upgrade handlers. 

I haven't added any actual handling/msgs since we currently don't want to do this via  on chain txn.

closes BLO-1387

### CODE REVIEW

File: x/marketmap/keeper/keeper.go [hunk 1]
```diff
@@ -118,6 +118,20 @@ func (k *Keeper) UpdateMarket(ctx sdk.Context, market types.Market) error {
...
+func (k *Keeper) DeleteMarket(ctx sdk.Context, tickerStr string) error {
+	// Check if Ticker exists
+	alreadyExists, err := k.markets.Has(ctx, types.TickerString(tickerStr))
+	if err != nil {
+		return err
+	}
+	if !alreadyExists {
+		return types.NewMarketDoesNotExistsError(types.TickerString(tickerStr))
+	}
+	// Create the config
```
Comment (technicallyty) [inline x/marketmap/keeper/keeper.go]:
```suggestion
```

**Review by technicallyty — COMMENTED** [review]
is there a possible error scenario here?

- market gets deleted
- somewhere down the line an authority wants to add a market back, which calls the `AfterMarketCreated` hook
- `AfterMarketCreated` fails saying CurrencyPair already exists, even though it was deleted from MM?

do we also need a AfterMarketDeleted hook?

**Review by aljo242 — APPROVED** [review]

**Comment by Eric-Warehime** [comment]
> is there a possible error scenario here?
> 
> * market gets deleted
> * somewhere down the line an authority wants to add a market back, which calls the `AfterMarketCreated` hook
> * `AfterMarketCreated` fails saying CurrencyPair already exists, even though it was deleted from MM?
> 
> do we also need a AfterMarketDeleted hook?

I think we don't want to manage that in hooks. I expect that the DeleteMarket calls will all only happen in upgrade handlers for now, and I would also expect that any x/oracle state should be cleaned up at the same time via `RemoveCurrencyPair` calls.

cc @technicallyty lmk if you agree or not.

**Comment by technicallyty** [comment]
> > is there a possible error scenario here?
> > 
> > * market gets deleted
> > * somewhere down the line an authority wants to add a market back, which calls the `AfterMarketCreated` hook
> > * `AfterMarketCreated` fails saying CurrencyPair already exists, even though it was deleted from MM?
> > 
> > do we also need a AfterMarketDeleted hook?
> 
> I think we don't want to manage that in hooks. I expect that the DeleteMarket calls will all only happen in upgrade handlers for now, and I would also expect that any x/oracle state should be cleaned up at the same time via `RemoveCurrencyPair` calls.
> 
> cc @technicallyty lmk if you agree or not.

sounds fine to me. maybe lets throw a comment on the method to warn callers to clean up x/oracle state?

**Review by technicallyty — APPROVED** [review]

**Pull request merged**